octoprint_mrbeam/comm_acc2.py

```diff
@@ -828,61 +1079,92 @@ def _gcode_S_sending(self, cmd, cmd_type=None):
 		cmd = self._replace_intensity(cmd)
 
 	def sendCommand(self, cmd, cmd_type=None, processed=False):
-		cmd = cmd.encode('ascii', 'replace')
-		if not processed:
-			cmd = process_gcode_line(cmd)
-			if not cmd:
-				return
-
-		if cmd[0] == "/":
-			self._log("Command: %s" % cmd)
-			specialcmd = cmd[1:].lower()
-			if "togglestatusreport" in specialcmd:
-				if self._status_timer is None:
-					self._status_timer = RepeatedTimer(1, self._poll_status)
+		if cmd is not None and cmd.strip().startswith('/'):
+			try:
+				cmd = cmd.strip()
+				self._log("Command: %s" % cmd)
+				self._logger.info("Terminal user command: %s", cmd)
+				tokens = cmd.split(' ')
+				specialcmd = tokens[0].lower()
+				if specialcmd.startswith('/togglestatusreport'):
+					if self._status_timer is None:
+						self._status_timer = RepeatedTimer(1, self._poll_status)
+						self._status_timer.start()
+					else:
+						self._status_timer.cancel()
+						self._status_timer = None
+				elif specialcmd.startswith('/setstatusfrequency'):
+					try:
+						frequency = float(tokens[1])
+					except ValueError:
+						self._log("No frequency setting found! Using 1 sec.")
+						frequency = 1
+					if self._status_timer is not None:
+						self._status_timer.cancel()
+					self._status_timer = RepeatedTimer(frequency, self._poll_status)
 					self._status_timer.start()
+				elif specialcmd.startswith('/disconnect'):
+					self.close()
+				elif specialcmd.startswith('/feedrate'):
+					if len(tokens) > 1:
+						self._set_feedrate_override(int(tokens[1]))
+					else:
+						self._log("no feedrate given")
+				elif specialcmd.startswith('/intensity'):
+					if len(tokens) > 1:
+						data = specialcmd[8:]
+						self._set_intensity_override(int(tokens[1]))
+					else:
+						self._log("no intensity given")
+				elif specialcmd.startswith('/reset'):
+					self._log("Reset initiated")
+					self._serial.write(list(bytearray('\x18')))
+				elif specialcmd.startswith('/flash_grbl'):
```

Comment:
check printer state. Only if operational, locked, closeddisconnected, closed_on_error

Reply:
in flash_grbl method